test/framework/nutanix.go

```diff
@@ -120,233 +125,267 @@ func RequiredNutanixEnvVars() []string {
-		name := os.Getenv(nutanixTemplateNameUbuntu127Var)
-		v.fillers = append(v.fillers, v.withUbuntuNutanixUUID(name)...)
-	}
+	return withNutanixKubeVersionAndOSForUUID(anywherev1.Kube127, Ubuntu2004, nil)
 }
 
-func (s *Nutanix) withUbuntuNutanixUUID(name string) []api.NutanixFiller {
-	uuid, err := s.client.GetImageUUIDFromName(context.Background(), name)
+func (n *Nutanix) withNutanixUUID(name string, osFamily anywherev1.OSFamily) []api.NutanixFiller {
+	uuid, err := n.client.GetImageUUIDFromName(context.Background(), name)
 	if err != nil {
-		s.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
+		n.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
 	}
 	return append([]api.NutanixFiller{},
 		api.WithNutanixMachineTemplateImageUUID(*uuid),
-		api.WithOsFamilyForAllNutanixMachines(anywherev1.Ubuntu),
+		api.WithOsFamilyForAllNutanixMachines(osFamily),
 	)
 }
 
 // WithPrismElementClusterUUID returns a NutanixOpt that adds API fillers to use a PE Cluster UUID.
 func WithPrismElementClusterUUID() NutanixOpt {
-	return func(v *Nutanix) {
+	return func(n *Nutanix) {
 		name := os.Getenv(nutanixPrismElementClusterName)
-		uuid, err := v.client.GetClusterUUIDFromName(context.Background(), name)
+		uuid, err := n.client.GetClusterUUIDFromName(context.Background(), name)
 		if err != nil {
-			v.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
+			n.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
 		}
-		v.fillers = append(v.fillers, api.WithNutanixPrismElementClusterUUID(*uuid))
+		n.fillers = append(n.fillers, api.WithNutanixPrismElementClusterUUID(*uuid))
 	}
 }
 
 // WithNutanixSubnetUUID returns a NutanixOpt that adds API fillers to use a Subnet UUID.
 func WithNutanixSubnetUUID() NutanixOpt {
-	return func(v *Nutanix) {
+	return func(n *Nutanix) {
 		name := os.Getenv(nutanixSubnetName)
-		uuid, err := v.client.GetSubnetUUIDFromName(context.Background(), name)
+		uuid, err := n.client.GetSubnetUUIDFromName(context.Background(), name)
 		if err != nil {
-			v.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
+			n.t.Fatalf("Failed to get UUID for image %s: %v", name, err)
 		}
-		v.fillers = append(v.fillers, api.WithNutanixSubnetUUID(*uuid))
+		n.fillers = append(n.fillers, api.WithNutanixSubnetUUID(*uuid))
 	}
 }
 
-// UpdateNutanixUbuntuTemplate123Var returns NutanixFiller by reading the env var and setting machine config's
-// image name parameter in the spec.
-func UpdateNutanixUbuntuTemplate123Var() api.NutanixFiller {
-	return api.WithNutanixStringFromEnvVar(nutanixTemplateNameUbuntu123Var, api.WithNutanixMachineTemplateImageName)
+// templateForKubeVersionAndOS returns a Nutanix filler for the given OS and Kubernetes version.
+func (n *Nutanix) templateForKubeVersionAndOS(kubeVersion anywherev1.KubernetesVersion, os OS, release *releasev1.EksARelease) api.NutanixFiller {
+	var template string
+	if release == nil {
+		template = n.templateForDevRelease(kubeVersion, os)
```

Comment:
yeah this ties to my previous comment about passing nil
Let's follow up on this to avoid having to do this default in all providers.

Reply:
Would you opposed to using variadics for this or perhaps a different method for tests where release is non-nil?

Reply:
will add this as a suggestion to the issue